### ISSUE TITLE
Conda 4.2.x may hang after install / update operations on Windows

### ISSUE THREAD

**Issue by pitrou** [conda/conda#3473]
Conda may try to clean up some trash as a background operation when running other commands, but it probably shouldn't sleep or wait doing so, in the event the cleaning fails.

**Comment by mparry** [comment]
I find this happens every time I run one of these operations. If I Ctrl+C during the lengthy post-update pause, the stack always looks like this:

```
Traceback (most recent call last):
  File "C:\Python27_x64\Scripts\conda-script.py", line 5, in <module>
    sys.exit(conda.cli.main())
  File "C:\Python27_x64\lib\site-packages\conda\cli\main.py", line 150, in main
    return conda_exception_handler(_main)
  File "C:\Python27_x64\lib\site-packages\conda\exceptions.py", line 473, in conda_exception_handler
    return_value = func(*args, **kwargs)
  File "C:\Python27_x64\lib\site-packages\conda\cli\main.py", line 144, in _main
    exit_code = args.func(args, p)
  File "C:\Python27_x64\lib\site-packages\conda\cli\main_update.py", line 66, in execute
    delete_trash()
  File "C:\Python27_x64\lib\site-packages\conda\common\disk.py", line 233, in delete_trash
    backoff_unlink(path)
  File "C:\Python27_x64\lib\site-packages\conda\common\disk.py", line 59, in backoff_unlink
    exp_backoff_fn(lambda f: lexists(f) and _unlink(f), file_or_symlink_path)
  File "C:\Python27_x64\lib\site-packages\conda\common\disk.py", line 164, in exp_backoff_fn
    sleep(sleep_time)
KeyboardInterrupt
```

Output from conda info:

```
C:\>conda info
Current conda install:

               platform : win-64
          conda version : 4.2.9
       conda is private : False
      conda-env version : 4.2.9
    conda-build version : 1.21.14+0.g4dfebe9.dirty
         python version : 2.7.12.final.0
       requests version : 2.11.1
       root environment : C:\Python27_x64  (writable)
    default environment : C:\Python27_x64
       envs directories : C:\Python27_x64\envs
          package cache : C:\Python27_x64\pkgs
           channel URLs : [...]
            config file : C:\Users\m.parry\.condarc
           offline mode : False
```

**Comment by mparry** [comment]
Seems to have been resolved under #3664
